### PR TITLE
DataCite plugin for import

### LINKED ISSUE #8564: DataCite import service
**Is your feature request related to a problem? Please describe.**
We cannot import metadata for DOI registered with DataCite.

**Describe the solution you'd like**
Initially we would like to import metadata from DataCite in the same way as from Crossref.

In a further commit we want to enable parallel data queries for multiple sources like DataCite and Crossref.

**Describe alternatives or workarounds you've considered**
We tried to access the desired metadata via Crossref, but that did not work. 
An example: '10.48550/ARXIV.2207.04779' can not be found in Crossref but in DataCite.

**Additional context**
PR to be opened.

### PR DESCRIPTION
**Pull request by johannastaudinger**
Plugin for import of external metadata from DataCite for research data and publications. Code by @johannastaudinger, @floriangantner and @philipprumpf.

## References
* Fixes #8564 

## Description
This is new code to enable an DataCite plugin for metadata import. Tests have been adjusted (but no new tests have been written so far) and configuration is required.

## Instructions for Reviewers
List of changes in this PR:
* dspace-api/pom.xml: Add JsonPath dependency
* dspace-api/src/main/resources/spring/spring-dspace-addon-import-services.xml: Configure DataCite plugin
* dspace-api/src/test/data/dspaceFolder/config/spring/api/external-services.xml: Add data provider for DataCite
* dspace-server-webapp/src/test/java/org/dspace/app/rest/ExternalSourcesRestControllerIT.java: Adjust test
* New files: 
   * dspace-api/src/main/java/org/dspace/importer/external/datacite/DataCiteFieldMapping.java: Implementation of field mapping for DataCite
   * dspace-api/src/main/java/org/dspace/importer/external/datacite/DataCiteImportMetadataSourceServiceImpl.java: Main class for plugin
   * dspace/config/spring/api/datacite-integration.xml: Main configuration for mapping

This PR includes just the minimal configuration. We can provide further configuration if requested.

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [ ] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods. -> Not sure.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide). -> Tests currently running!
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation. -> Only copied dependency from main pom.xml.

### CODE REVIEW

**Comment by johannastaudinger** [comment]
Tests just completed successfully for: `mvn install -DskipUnitTests=false -DskipIntegrationTests=false`.

**Comment by tdonohue** [comment]
@johannastaudinger : I'd recommend removing usage of GSon in this PR (as we attempted to remove it entirely from our codebase here #8259).  

I'd also recommend migrating to using `org.json.*` classes instead of JSONPath (as that makes this code easier to maintain alongside other import tools -- obviously if different import tools use different JSON librarys, that makes maintenance more complex).  For the same reason, it'd be better if you aligned this code to use `LiveImportClient`, as that again aligns it with other import tools/classes.  In general, we highly recommend using similar libraries to other tools... the fewer libraries DSpace uses, the easier it is to maintain.

While I'm adding some notes here, I'll also add that this PR looks to be missing ITs / tests.  Obviously, we'd want those to be created (even with mock data) to help prove the code is working as expected (and again, tests make code easier to maintain as we can more easily tell when we've broken a feature).

**Comment by johannastaudinger** [comment]
Thanks for the reply!

The reason we used these libraries is that the code was created some time ago when we were working with an older DSpace-CRIS version as our basis. 
We will remove GSon and JSONPath, and try to implement `org.json.*` as well as `LiveImportClient` following your advice.

Tests will be coming soon, we are currently working on them. 

Thank you for the consideration of our contribution for the 7.5 release, we will try our best to complete everything in the next few days!

File: dspace-api/src/main/java/org/dspace/importer/external/datacite/DataCiteImportMetadataSourceServiceImpl.java
Comment (floriangantner) [inline dspace-api/src/main/java/org/dspace/importer/external/datacite/DataCiteImportMetadataSourceServiceImpl.java:57]: still some legacy and unused import

Comment (johannastaudinger) [reply dspace-api/src/main/java/org/dspace/importer/external/datacite/DataCiteImportMetadataSourceServiceImpl.java:57]: Thanks @floriangantner, will be fixed asap!

**Comment by johannastaudinger** [comment]
This version uses `com.fasterxml.jackson` but does not use `org.json.*`, just as in `CrossRefImportMetadataSourceServiceImpl.java`. Is that okay?

Tests have been expanded slightly and we fixed the issue @floriangantner mentioned.

File: dspace-api/src/main/java/org/dspace/importer/external/datacite/DataCiteImportMetadataSourceServiceImpl.java
Comment (johannastaudinger) [reply dspace-api/src/main/java/org/dspace/importer/external/datacite/DataCiteImportMetadataSourceServiceImpl.java]: Has been fixed! :grin:

File: dspace-api/src/main/java/org/dspace/importer/external/datacite/DataCiteImportMetadataSourceServiceImpl.java
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/importer/external/datacite/DataCiteImportMetadataSourceServiceImpl.java:54]: I would recommend making this a configuration in the `dspace/config/modules/external-providers.cfg` file (as most other import tools also have their URL configurable in that location). It might look something like:
```
datacite.url = https://api.datacite.org/dois/
```

This allows sites to easily change this URL if it ever changes for DataCite.

File: dspace-api/src/test/data/dspaceFolder/config/spring/api/external-services.xml
Comment (tdonohue) [inline dspace-api/src/test/data/dspaceFolder/config/spring/api/external-services.xml:94]: This same bean needs to be added to the default `dspace/config/spring/api/external-services.xml` file.  Without it, I couldn't get this PR to work correctly.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@johannastaudinger : Thanks!  The new changes look good overall.  I tested this (after having to manually modify my `external-sources.xml` -- see inline note below).  I've verified that it works when searching either by a DOI or by a topic/term.  Either way, I'm getting results from the MyDSpace -> Import metadata from external source.  I also verified I can successfully create a new Submission after selecting one of the results.

A few minor suggestions inline below. But, overall, I think this looks good.

SIDENOTE: After this is merged, we'll need to add a new `submission.import-external.source.datacite` i18n key to the DSpace User Interface's `src/assets/i18n/en.json5` translation file.  This would just ensure that this import tools shows up as "DataCite" instead of displaying that long key.  Either I can add it, or you could create a small PR (once this PR is fully ready).

**Comment by johannastaudinger** [comment]
@tdonohue : Thank you for your review! We tried to implement your recommendations. Does this look okay?

**Review by tdonohue — APPROVED** [review]
👍 Thanks @johannastaudinger (and all others involved)!  This looks good to me now and it's working perfectly.  I can search both by a title and by a DOI in DataCite, and when I import the Item from DataCite, I see the metadata brought over (including the DOI into `dc.identifier.doi`).  So, this all seems to be working as described.

I'll create a tiny PR to add the missing `submission.import-external.source.datacite` i18n key to the Angular UI (so that it's not accidentally forgotten), as that seems like the only thing missing.

**Comment by tdonohue** [comment]
Merging with +1 approval.  The i18n key PR was created at https://github.com/DSpace/dspace-angular/pull/2038   I'll merge that PR shortly as well (once it passes our tests/CI checks).

**Pull request merged**
